ffi/src/rdcleanpath.rs

```diff
@@ -0,0 +1,238 @@
+        pub fn new_request(
+            x224_pdu: &[u8],
+            destination: &str,
+            proxy_auth: &str,
+            pcb: &str,
+        ) -> Result<Box<RDCleanPathPdu>, Box<IronRdpError>> {
+            let pcb_opt = if pcb.is_empty() { None } else { Some(pcb.to_owned()) };
+
+            let pdu = ironrdp_rdcleanpath::RDCleanPathPdu::new_request(
+                x224_pdu.to_vec(),
+                destination.to_owned(),
+                proxy_auth.to_owned(),
+                pcb_opt,
+            )
+            .context("failed to create RDCleanPath request")
+            .map_err(GenericError)?;
+
+            Ok(Box::new(RDCleanPathPdu(pdu)))
+        }
+
+        /// Decodes a RDCleanPath PDU from DER-encoded bytes
+        pub fn from_der(bytes: &[u8]) -> Result<Box<RDCleanPathPdu>, Box<IronRdpError>> {
+            let pdu = ironrdp_rdcleanpath::RDCleanPathPdu::from_der(bytes)
+                .context("failed to decode RDCleanPath PDU")
+                .map_err(GenericError)?;
+
+            Ok(Box::new(RDCleanPathPdu(pdu)))
+        }
+
+        /// Encodes the RDCleanPath PDU to DER-encoded bytes
+        pub fn to_der(&self) -> Result<Box<VecU8>, Box<IronRdpError>> {
+            let bytes = self
+                .0
+                .to_der()
+                .context("failed to encode RDCleanPath PDU")
+                .map_err(GenericError)?;
+
+            Ok(Box::new(VecU8(bytes)))
+        }
+
+        /// Detects if the bytes contain a valid RDCleanPath PDU and returns detection result
+        pub fn detect(bytes: &[u8]) -> Box<RDCleanPathDetectionResult> {
+            let result = ironrdp_rdcleanpath::RDCleanPathPdu::detect(bytes);
+            Box::new(RDCleanPathDetectionResult(result))
+        }
+
+        /// Gets the type of this RDCleanPath PDU
+        pub fn get_type(&self) -> Result<RDCleanPathResultType, Box<IronRdpError>> {
+            let rdcleanpath = self
+                .0
+                .clone()
+                .into_enum()
+                .context("missing RDCleanPath field")
+                .map_err(GenericError)?;
+
+            let result_type = match rdcleanpath {
+                ironrdp_rdcleanpath::RDCleanPath::Request { .. } => RDCleanPathResultType::Request,
+                ironrdp_rdcleanpath::RDCleanPath::Response { .. } => RDCleanPathResultType::Response,
+                ironrdp_rdcleanpath::RDCleanPath::GeneralErr(_) => RDCleanPathResultType::GeneralError,
+                ironrdp_rdcleanpath::RDCleanPath::NegotiationErr { .. } => RDCleanPathResultType::NegotiationError,
+            };
+
+            Ok(result_type)
+        }
+
+        /// Gets the X.224 connection response bytes (for Response or NegotiationError variants)
+        pub fn get_x224_response(&self) -> Result<Box<VecU8>, Box<IronRdpError>> {
+            let rdcleanpath = self
+                .0
+                .clone()
+                .into_enum()
```

Comment:
issue: You don’t need to use `into_enum()` at all. You can access the properties directly.